### PR TITLE
ENG-5724: Add max idle and open connections settings

### PR DESCRIPTION
**Pull request by Darkneon**
https://alpaca.atlassian.net/browse/ENG-5724

**Background:**
We are in the process of adding the ability to configure max idle and open connection in gotrader. 

**Changes:**
Add max idle and open connections as setting so gotrader can pass them

### CODE REVIEW

File: sqlstore.go [hunk 1]
```diff
@@ -48,23 +54,44 @@ func (f sqlStoreFactory) Create(sessionID SessionID) (msgStore MessageStore, err
 			return nil, err
 		}
 	}
-	return newSQLStore(sessionID, sqlDriver, sqlDataSourceName, sqlConnMaxLifetime)
+	sqlConnMaxIdle := 100
```
Comment (Darkneon) [inline sqlstore.go]: Using default value from OMS

File: sqlstore.go [hunk 2]
```diff
@@ -48,23 +54,44 @@ func (f sqlStoreFactory) Create(sessionID SessionID) (msgStore MessageStore, err
...
+	if sessionSettings.HasSetting(config.SQLStoreConnMaxIdle) {
+		sqlConnMaxIdle, err = sessionSettings.IntSetting(config.SQLStoreConnMaxIdle)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sqlConnMaxOpen := 100
```
Comment (Darkneon) [inline sqlstore.go]: Using default value from OMS

File: sqlstore.go (hunk 1 above)
Comment (neal) [reply sqlstore.go]: this is too much for gotrader. 10/20 for idle/open should be good enough default.

Comment (Darkneon) [reply sqlstore.go]: Lowered it to 10. This is more for a default value if you omit passing the settings. I also updated gotrader to pass a value when creating a new fix client

Comment (neal) [reply sqlstore.go]: Can you also update the max open conns?

Comment (Darkneon) [reply sqlstore.go]: Yes, done

File: sqlstore.go [hunk 3]
```diff
@@ -48,23 +54,44 @@ func (f sqlStoreFactory) Create(sessionID SessionID) (msgStore MessageStore, err
...
 
-func newSQLStore(sessionID SessionID, driver string, dataSourceName string, connMaxLifetime time.Duration) (store *sqlStore, err error) {
+func newSQLStore(sessionID SessionID, driver string, dataSourceName string, dbs dbSettings) (store *sqlStore, err error) {
 	store = &sqlStore{
 		sessionID:          sessionID,
 		cache:              &memoryStore{},
 		sqlDriver:          driver,
 		sqlDataSourceName:  dataSourceName,
-		sqlConnMaxLifetime: connMaxLifetime,
+		sqlConnMaxLifetime: dbs.connMaxLifetime,
```
Comment (neal) [inline sqlstore.go]: can `sqlConnMaxLifetime` be removed from `sqlStore` - does not look like it's being used anymore?

Comment (neal) [reply sqlstore.go]: maybe you can keep `dbSettings` in there

Comment (Darkneon) [reply sqlstore.go]: That implementation is coming from upstream. We are OK with diverging?

Comment (neal) [reply sqlstore.go]: sure, we're already pretty diverged

Comment (Darkneon) [reply sqlstore.go]: Done

**Review by neal — APPROVED** [review]

**Pull request merged**